tests/Fixer/PhpUnit/PhpUnitDataProviderReturnTypeFixerTest.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\PhpUnit;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\PhpUnit\PhpUnitDataProviderReturnTypeFixer
+ */
+final class PhpUnitDataProviderReturnTypeFixerTest extends AbstractFixerTestCase
+{
+    private const TEMPLATE = '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFooCases
+     */
+    public function testFoo() {}
+    /**
+     * @dataProvider provider
+     */
+    public function testBar() {}
+    public function provideFooCases()%s {}
+    public function provider()%s {}
+    public function notProvider(): array {}
+}';
+
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{0: string, 1?: string}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield 'data provider with iterable return type' => [
+            '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFooCases
+     */
+    public function testFoo() {}
+    public function provideFooCases() : iterable {}
+}',
+        ];
+
+        yield 'data provider without return type' => self::mapToTemplate(
+            ': iterable',
+            '',
+        );
+
+        yield 'data provider with array return type' => self::mapToTemplate(
+            ': iterable',
+            ': array',
+        );
+
+        yield 'data provider with return type and comment' => self::mapToTemplate(
+            ': /* foo */ iterable',
+            ': /* foo */ array',
+        );
+
+        yield 'data provider with return type namespaced class' => self::mapToTemplate(
+            ': iterable',
```

Comment:
Consistency is king, right? `Foo\Bar` must be iterable.

Reply:
But isn't it static analyser's responsibility to ensure that defined return type actually _is_ `iterable`? It can be implicitly iterable: `class Bar implements IteratorAggregate` (explicitly iterable like `Foo\Bar&iterable` is [not possible](https://3v4l.org/U3XF8)). `Foo\Bar` may have generic annotations and provide information about iterated items, this would be lost. But since it's risky, maybe it's OK... 🤷‍♂️

TL;DR: what I mean is maybe this fixer should only replace `array` type or add `iterable` if no return type is defined. But then it's not risky anymore (probably) 😅.

Reply:
I wonder if this is the same level o "in theory" situation as we had the other day regarding `ConfigInterface`, the test case was added more to show how it behaves, not to ensure it behaves that way. Have you ever seen that a custom class was returned from the data provider?

What would you say we go with the current "everything to iterable" approach and if anyone reports "hey, may AwesomeDataProvider class type is changed to iterable" then we add "exclude" config option or something like that?

Reply:
Of course, it's an edge case (I have never seen something in my life 😉), we can do like you wrote.